src/main/java/school/hei/haapi/endpoint/rest/controller/FeeController.java

```diff
@@ -40,6 +37,21 @@ public List<Fee> createFees(
         .collect(toUnmodifiableList());
   }
 
+  @PutMapping("/students/{studentId}/fees")
+  public List<Fee> updateStudentFees(@PathVariable String studentId, @RequestBody List<Fee> fees) {
+    List<school.hei.haapi.model.Fee> domainFeeList =
+        fees.stream()
+            .map(
+                fee -> {
+                  User student = userService.getById(fee.getStudentId());
+                  return feeMapper.toDomain(fee, student);
+                })
```

Comment:
dubious.

first, add a validator which will verify whether all Fees in the requestBody are for the same studentId.
secondly, get the Student only once for a better performance.
Then map the List of fees with the student.

NB : do not use peek() method even if if is recommended by your IDE